### PR TITLE
URL Navigation

### PR DESCRIPTION
**Pull request by kn-ms**
## Bug / Requirement Description
The url on teh ui does not preserv navigation state

## Solution description
url routing on the ui is now the source of navigation state

## Checklist:
- [x] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/web_ui/testing/src/Report/BatchReport.js [hunk 1]
```diff
@@ -73,9 +73,12 @@ class BatchReport extends React.Component {
       report: processedReport,
       filteredReport,
       selectedUIDs,
-      lastManualSelectedUIDs: selectedUIDs,
       loading: false,
       logs: report.logs,
+    }, () => {
+      this.props.history.replace(
```
Comment (Pyifan) [inline testplan/web_ui/testing/src/Report/BatchReport.js:79]: What is this history used for?

**Review by Pyifan — COMMENTED** [review]
Like this change!

File: testplan/web_ui/testing/src/Report/BatchReport.js (hunk 1 above)
Comment (kn-ms) [reply testplan/web_ui/testing/src/Report/BatchReport.js:79]: history representiung the browser history, it is injected by the router. If you push or replace things in it that made the browser to actuially navigate. Push equivalent of redirect while replace is similar, just the original url is removed from history and replaced with the new one.

**Comment by kn-ms** [comment]
I will close and open a new one so the migrated travis job is picked up correctly

**Pull request closed without merging**